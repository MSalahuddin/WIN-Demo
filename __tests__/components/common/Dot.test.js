import React from 'react';
import { render } from 'react-native-testing-library';

import DotComponent from '../../../js/components/common/Dot';

describe('Component Dot', () => {
  test('renders correctly with default props', () => {
    const output = render(<DotComponent />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly when active', () => {
    const output = render(<DotComponent isActive />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly when not active', () => {
    const output = render(<DotComponent isActive={false} />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly with different size', () => {
    const output = render(<DotComponent size={20} />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly with different color', () => {
    const output = render(<DotComponent dotColor="#000" />);
    expect(output).toMatchSnapshot();
  });
});
