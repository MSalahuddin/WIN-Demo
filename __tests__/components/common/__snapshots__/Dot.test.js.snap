// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Dot renders correctly when active 1`] = `
<View
  accessible={true}
  dotColor="#FFFFFF"
  focusable={true}
  isActive={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={12}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "height": 24,
        "marginHorizontal": 4,
        "opacity": 1,
        "shadowColor": "#FFFFFF",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
        "width": 24,
      },
    ]
  }
/>
`;

exports[`Component Dot renders correctly when not active 1`] = `
<View
  accessible={true}
  dotColor="#FFFFFF"
  focusable={true}
  isActive={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={12}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 9,
        "height": 18,
        "marginHorizontal": 4,
        "opacity": 0.5,
        "shadowColor": "#FFFFFF",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
        "width": 18,
      },
    ]
  }
/>
`;

exports[`Component Dot renders correctly with default props 1`] = `
<View
  accessible={true}
  dotColor="#FFFFFF"
  focusable={true}
  isActive={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={12}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 9,
        "height": 18,
        "marginHorizontal": 4,
        "opacity": 0.5,
        "shadowColor": "#FFFFFF",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
        "width": 18,
      },
    ]
  }
/>
`;

exports[`Component Dot renders correctly with different color 1`] = `
<View
  accessible={true}
  dotColor="#000"
  focusable={true}
  isActive={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={12}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "borderRadius": 9,
        "height": 18,
        "marginHorizontal": 4,
        "opacity": 0.5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
        "width": 18,
      },
    ]
  }
/>
`;

exports[`Component Dot renders correctly with different size 1`] = `
<View
  accessible={true}
  dotColor="#FFFFFF"
  focusable={true}
  isActive={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={20}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 15,
        "height": 30,
        "marginHorizontal": 6.666666666666667,
        "opacity": 0.5,
        "shadowColor": "#FFFFFF",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
        "width": 30,
      },
    ]
  }
/>
`;
